langlib/lang.xml/src/main/java/org/ballerinalang/langlib/xml/Filter.java

```diff
@@ -57,7 +57,7 @@ public static BXml filter(BXml x, BFunctionPointer<Object, Boolean> func) {
             func.asyncCall(args,
                       result -> {
                           if ((Boolean) result) {
-                              return ValueCreator.createXmlSequence(x);
+                              return x;
```

Comment:
Is this correct? My concern is that the specification mentions that the `filter` function returns a new XML sequence containing items in parameter `x` for which function `func` evaluates to true. So, the value returned from the `filter` function should be an XML sequence. If that is the case, we need to fix this in the `isEqual` function in the type checker, where we check the tag of the type.

Reply:
Under the [sequence value section](https://ballerina.io/spec/lang/master/#section_5.3) in the specification it is defined that, 
> Note that for a sequence consisting of a single item v is the same thing as v. A single item is a sequence.

Thus IMO it is correct here not to create a sequence.

Reply:
@MaryamZi What is your opinion about this? 
`Note that for a sequence consisting of a single item, v is the same thing as v. A single item is a sequence.`
With this also, I feel that a single item should be treated as a sequence.

Reply:
From a user PoV, there's no difference. You can't differentiate between a sequence of length 1 and a singleton in Ballerina code. The implementation could choose either way as long as sequences of length 1 are handled correctly everywhere. I would probably go with not creating a sequence.

Reply:
Got it. Thanks for the clarification, @poorna2152 So, we can keep the logic as it is now.

Reply:
Ack